pkg/pipeline/utils/timed_cache.go

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package utils
+
+import (
+	"container/list"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Functions to manage an LRU cache with an expiry
+// When an item expires, allow a callback to allow the specific implementation to perform its particular cleanup
+
+type CacheCallback interface {
+	Cleanup(entry interface{})
+}
+
+type cacheEntry struct {
+	key             string
+	lastUpdatedTime int64
+	e               *list.Element
+	SourceEntry     interface{}
+}
+
+type TimedCacheMap map[string]*cacheEntry
+
+type TimedCache struct {
+	Mu        sync.Mutex
+	CacheList *list.List
+	CacheMap  TimedCacheMap
+}
+
+func (l *TimedCache) GetCacheEntry(key string) (interface{}, bool) {
```

Comment:
Why the identifier of the receiver of all the methods of `*TimedCache` is `l`? (and not `t` or `tc`...)